src/core/mesh/qgsmeshlayerelevationproperties.h

```diff
@@ -70,12 +72,44 @@ class CORE_EXPORT QgsMeshLayerElevationProperties : public QgsMapLayerElevationP
      */
     void setProfileLineSymbol( QgsLineSymbol *symbol SIP_TRANSFER );
 
+    /**
+     * Returns the fill symbol used to render the mesh profile in elevation profile plots.
+     *
+     * \see setProfileFillSymbol()
+     */
+    QgsFillSymbol *profileFillSymbol() const;
+
+    /**
+     * Sets the fill \a symbol used to render the mesh profile in elevation profile plots.
+     *
+     * Ownership of \a symbol is transferred to the plot.
+     *
+     * \see profileFillSymbol()
+     */
+    void setProfileFillSymbol( QgsFillSymbol *symbol SIP_TRANSFER );
+
+    /**
+     * Returns the symbology option used to render the mesh profile in elevation profile plots.
+     *
+     * \see setProfileSymbology()
+     */
+    Qgis::ProfileSurfaceSymbology profileSymbology() const { return mSymbology; }
+
+    /**
+     * Sets the \a symbology option used to render the mesh profile in elevation profile plots.
+     *
+     * \see setProfileSymbology()
+     */
+    void setProfileSymbology( Qgis::ProfileSurfaceSymbology symbology );
+
   private:
 
-    void setDefaultProfileLineSymbol();
+    void setDefaultProfileLineSymbol( const QColor &color );
```

Comment:
```suggestion
    void setDefaultProfileLineSymbolColor( const QColor &color );
```
?

Reply:
Ah it's doing more than just setting the color of the symbol though -- it's creating a brand new symbol with that color.